qutebrowser/completion/completiondelegate.py

```diff
@@ -197,9 +196,20 @@ def _get_textdoc(self, index):
             pattern = index.model().pattern
             columns_to_filter = index.model().srcmodel.columns_to_filter
             if index.column() in columns_to_filter and pattern:
-                repl = r'<span class="highlight">\g<0></span>'
-                text = re.sub(re.escape(pattern), repl, self._opt.text,
-                              flags=re.IGNORECASE)
+                if config.get('completion', 'match-type') == 'fuzzy':
+                    text = ''
+                    for char in self._opt.text:
+                        if char in pattern:
+                            text += '<span class="highlight">%s</span>' % char
+                        else:
+                            text += char
+                else:
+                    repl = r'<span class="highlight">\g<0></span>'
```

Comment:
Note to self: This should probably be HTML-escaped correctly - but that's not an issue with your PR, as it was already wrong before.